asllib/doc/AssignableExpressions.tex

```diff
@@ -61,17 +61,51 @@ \chapter{Assignable Expressions\label{chap:AssignableExpressions}}
 
 \section{Syntax\label{sec:AssignableExpressionsSyntax}}
 \begin{flalign*}
-\Nlexpr \derives\ & \Nlexpratom &\\
-|\ & \Tminus &\\
-|\ & \Tlpar \parsesep \Clisttwo{\Nlexpr} \parsesep \Trpar &\\
-\Nlexpratom \derives\ & \Tidentifier &\\
-|\ & \Nlexpratom \parsesep \Nslices &\\
-|\ & \Nlexpratom \parsesep \Tllbracket \parsesep \Nexpr \parsesep \Trrbracket &\\
-|\ & \Nlexpratom \parsesep \Tdot \parsesep \Tidentifier{\field} &\\
-|\ & \Nlexpratom \parsesep \Tdot \parsesep \Tlbracket \parsesep \Clist{{\Tidentifier}} \parsesep \Trbracket &
+\Nlexpr \derives\
+   & \Tminus &\\
+|\ & \Nslicedbasiclexpr &\\
+|\ & \Tlpar \parsesep \Clisttwo{\Ndiscardorslicedbasiclexpr} \parsesep \Trpar &\\
+|\ & \Tidentifier \parsesep \Tdot \parsesep \Tlbracket \parsesep \Clisttwo{\Tidentifier} \parsesep \Trbracket &\\
+|\ & \Tidentifier \parsesep \Tdot \parsesep \Tlpar \parsesep \Clisttwo{\Nignoredoridentifier} \parsesep \Trpar &
+\end{flalign*}
+
+\begin{flalign*}
+\Nbasiclexpr \derives\
+   & \Tidentifier \parsesep \Nnestedfields &\\
+|\ & \Tidentifier \parsesep \Tllbracket \parsesep \Nexpr \parsesep \Trrbracket \parsesep \Nnestedfields &
+\end{flalign*}
+
+\begin{flalign*}
+\Nnestedfields \derives\ & \emptysentence \;|\; \Tdot \parsesep \Tidentifier \parsesep \Nnestedfields &
+\end{flalign*}
+
+\begin{flalign*}
+\Nslicedbasiclexpr \derives\ & \Nbasiclexpr \;|\; \Nbasiclexpr \parsesep \Nslices &
+\end{flalign*}
+
+\begin{flalign*}
+\Ndiscardorslicedbasiclexpr \derives\ & \Tminus \;|\; \Nslicedbasiclexpr &
 \end{flalign*}
 
 \subsection{Abstract Syntax Builders\label{sec:AssignableExpressionsAbstractSyntaxBuilders}}
+
+We rely on the following definition in this section, which is used to represent a $\Nslicedbasiclexpr$.
```

Comment:
Is this an addition to the AST or a form of IR constructed from the AST? I'm guessing the latter. I think the intended role of this data type needs to be explained.

Also, there are now quite a few functions to desugar left-hand sides and it is easy to miss what's going on. Perhaps this deserves a section on its own. Something like "Desugaring Assignable Expressions" that provides an overview of what is the intent here. This can also be postpone to a lter PR.

Reply:
I've added some explanation, and had a go at introducing the "Desugaring Assignable Expressions" section - we can iterate on this in future PRs as necessary.